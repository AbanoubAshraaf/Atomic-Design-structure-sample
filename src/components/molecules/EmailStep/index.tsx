import React, { useState } from "react";
import InputField from "@/components/atoms/InputField";
import Button from "@/components/atoms/Button";

interface EmailStepProps {
  onClick: (field: string, value: string) => void;
}

const EmailStep: React.FC<EmailStepProps> = ({ onClick }) => {
  const [email, setEmail] = useState<string>("");
  const [error, setError] = React.useState<string>("");

  return (
    <div className="flex flex-col gap-4 w-full">
      <InputField
        label="Email"
        value={email}
        type="email"
        error={error}
        onChange={(value) => setEmail(value)}
      />
      <Button text="Next" onClick={() => onClick("email", email)} />
    </div>
  );
};

export default EmailStep;
