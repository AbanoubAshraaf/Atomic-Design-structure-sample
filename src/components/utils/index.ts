export const isValidAge = (age:number): boolean => {
    return (
      age > 0 && // Age must be positive
      !isNaN(age) && // Age must be a valid number
      Number.isInteger(age) // Age must be an integer
    );
  };
  