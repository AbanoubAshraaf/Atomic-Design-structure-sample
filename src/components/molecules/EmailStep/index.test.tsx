import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EmailStep from "@/components/molecules/EmailStep";
import { mockValidUserData } from "@/components/test/mocks";
import { createSnapShot } from "@/components/test/utils";

describe("<EmailStep/> Component tests", () => {
  test("render email input properly", () => {
    const { container, getByTestId } = render(<EmailStep onClick={jest.fn()} />);
    expect(getByTestId("email")).toBeInTheDocument();

    createSnapShot(container)
  });

  test("display an error for invalid email", () => {
    const { container, getByTestId, getByText } = render(<EmailStep onClick={jest.fn()} />);
    const input = getByTestId('email');
    fireEvent.change(input, { target: { value: 'invalid-email' } });
    fireEvent.click(getByText("Next"));
    expect(getByText("Please enter a valid email address.")).toBeInTheDocument();

    createSnapShot(container)
  });

  test("call callback with email value on button click", () => {
    const mockCallback = jest.fn();
    const { getByTestId, getByText } = render(<EmailStep onClick={mockCallback} />);

    fireEvent.change(getByTestId("email"), { target: { value: mockValidUserData.email } });
    fireEvent.click(getByText("Next"));
    expect(mockCallback).toHaveBeenCalledWith("email", "test@getsafe.com");
  });
});
