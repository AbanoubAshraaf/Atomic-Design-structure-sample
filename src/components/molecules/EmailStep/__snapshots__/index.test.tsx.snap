// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailStep/> Component tests call callback with email value on button click 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <label>
        Email
        :
      </label>
      <input
        class="input input-bordered input-primary w-full max-w-xs"
        data-testid="email"
        type="email"
        value="test@getsafe.com"
      />
    </div>
    <button
      class="btn btn-primary"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`<EmailStep/> Component tests display an error for invalid email 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <label>
        Email
        :
      </label>
      <input
        class="input input-bordered input-primary w-full max-w-xs"
        data-testid="email"
        type="email"
        value="invalid-email"
      />
      <span
        class="error-text"
      >
        Please enter a valid email address.
      </span>
    </div>
    <button
      class="btn btn-primary"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`<EmailStep/> Component tests render email input properly 1`] = `
<div>
  <div
    class="flex flex-col gap-4 w-full"
  >
    <div
      class="flex flex-col gap-2 items-start"
    >
      <label>
        Email
        :
      </label>
      <input
        class="input input-bordered input-primary w-full max-w-xs"
        data-testid="email"
        type="email"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
    >
      Next
    </button>
  </div>
</div>
`;
